File: SkillBridgeSmsService/src/main/java/com/skillBridge/sms_service/service/ProjectService.java
package com.skillBridge.sms_service.service;

import java.util.List;

import com.skillBridge.sms_service.dtos.ModifyProjectMembersRequest;
import com.skillBridge.sms_service.dtos.ProjectCreateRequest;
import com.skillBridge.sms_service.dtos.ProjectPatchRequest;
import com.skillBridge.sms_service.dtos.ProjectResponse;
import com.skillBridge.sms_service.entities.ProjectStatus;

public interface ProjectService {

    // ================= CREATE =================
    ProjectResponse create(ProjectCreateRequest request, Long studentId);

    // ================= READ (PUBLIC) =================
    ProjectResponse get(Long id);
    List<ProjectResponse> getAll();

    // ================= UPDATE =================
    ProjectResponse update(
            Long id,
            ProjectCreateRequest request,
            Long studentId
    );

    ProjectResponse patch(
            Long id,
            ProjectPatchRequest request,
            Long studentId
    );

    ProjectResponse updateStatus(
            Long id,
            ProjectStatus status,
            Long studentId
    );

    // ================= DELETE =================
    void delete(Long id, Long studentId);
    
    ProjectResponse modifyProjectMembers(
            Long projectId,
            ModifyProjectMembersRequest request,
            Long requesterId
    );

	boolean isMember(Long projectId, Long studentId);

}

File: SkillBridgeFrontend/src/services/projectService.js
import api from './api';

export const projectService = {
    getAll: async () => {
        const response = await api.get('/projects');
        return response.data;
    },

    getById: async (id) => {
        const response = await api.get(`/projects/${id}`);
        return response.data;
    },

    create: async (data) => {
        const response = await api.post('/projects/create', data);
        return response.data;
    },

    update: async (id, data) => {
        const response = await api.put(`/projects/update/${id}`, data);
        return response.data;
    },

    patch: async (id, data) => {
        const response = await api.patch(`/projects/patch/${id}`, data);
        return response.data;
    },

    updateStatus: async (id, status) => {
        const response = await api.patch(`/projects/${id}/status`, { status });
        return response.data;
    },

    delete: async (id) => {
        const response = await api.delete(`/projects/delete/${id}`);
        return response.data;
    },

    getApplications: async (id) => {
        const response = await api.get(`/projects/${id}/applications`);
        return response.data;
    },

    modifyMembers: async (id, data) => {
        const response = await api.patch(`/projects/${id}/members`, data);
        return response.data;
    },
};

export default projectService;
